### PR TITLE
Feature/finish migration to yarn

### PR DESCRIPTION
**Pull request by bsutt123**
This takes @jnimety work from #33 and finishes the process by...

1. changing jekylls configuration allow inclusion of `node_modules`
2. changing reference from a static file in a js folder to instead refernce the node modules in the jekyll template for `head` and `footer` for the most part.

Other issues and concerns related to dependency and package management.

1. Webcomponents isn't installed through yarn because there doesn't appear to be a npm package for the version we are using. CDN network is used instead.
2. Modernizr to check for flexbox functionality, but in browsers of today we can likely assume support for flexbox. In another Issue/PR we should remove the use in the css and then eliminate this direct dependency

I don't know if there is a way for me to put this at the bottom of #33, if I should have submitted some other way then I'm happy to change it. I was also not sure if I should rebase this branch so that it was concurrent with master, I didn't know if thats better to do on the merge or on my local machine. If I should rebase this branch with upstream master then let me know and I'll work on that next.

### CODE REVIEW

**Comment by jnimety** [comment]
I closed #33 in favor of this PR, thanks @bsutt123 ! Yes, please rebase against master and I think we can add `yarn-error.log` to `.gitignore`. Otherwise I think we can merge.

**Comment by bsutt123** [comment]
Conflicts removed with git rebase. I actually used Max's blog post to do it. Learning is the best.

File: _config.yml [hunk 1]
```diff
@@ -18,6 +18,11 @@ permalink: /blog/:title/
 paginate: 9,
 paginate_path: "/blog/page:num/"
 
+exclude: [
```
Comment (jnimety) [inline _config.yml]: when we implement webpack this goes away

Comment (bsutt123) [reply _config.yml]: Yeah, at that point the bundle should include what we need.

**Pull request merged**